daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -1806,68 +1881,101 @@ object VerifyTestCase {
    * @param expectedDiags                             Expected diagnostics from test cases
    * @param implString                                Implementation string
    */
-  def verifyAllDiagnosticsFound(
+  def verifyDiagnosticsFound(
     actualDiags: Seq[Diagnostic],
-    expectedDiags: Option[ErrorWarningBase],
+    expectedDiags: Option[Seq[ErrorWarningBase]],
+    ignoreUnexpectedWarnings: Boolean,
     implString: Option[String]
-  ) = {
+  ): Option[Seq[Unit]] = {
```

Comment:
Can this just return a `Unit`? I think this function either throws an exception if there are problems or returns nothing?